Add "Initialize User Secrets" menu item

seclerp opened the pull request:
The PR adds additional menu item to initialize user secrets:

![image](https://user-images.githubusercontent.com/20597871/124981773-9f4ab680-e03e-11eb-88d7-2e0c5de14f20.png)

"Initialize User Secrets" action executes `dotnet user-secrets init` on a supported project file.

I also refactored the current code a bit and moved a generic code into a dedicated service and extensions.

Manually tested according to repository guidelines.

Closes #6

seclerp commented:
@Witik could you take a look, please?

Witik commented on src/main/kotlin/eu/gillissen/rider/usersecrets/InitUserSecretsAction.kt:
```diff
@@ -0,0 +1,72 @@
+    override fun update(actionEvent: AnActionEvent) {
+        actionEvent.presentation.isEnabledAndVisible = false
+
+        if (!isActionSupported(actionEvent)) {
+            return
+        }
+
+        actionEvent.presentation.isVisible = true
+
+        // If Project already contains <UserSecretsId>
+        if (!actionEvent.getXmlUserSecretsIdValue().isNullOrEmpty()) {
+            return
+        }
```
I don't think it makes sense to show the action if the project already has user secrets.

```suggestion
        if (!isActionSupported(actionEvent) || !actionEvent.getXmlUserSecretsIdValue().isNullOrEmpty()) {
            return
        }
```

Witik replied:
Or better yet, add it to the `isActionSupported` method.

seclerp replied:
The idea behind disabled but visible is just about verbosity. In such a case the user will know that he has such ability to initialize secrets, but not for the selected project. With both disabled + not visible in such a case there is no way to check if the plugin not working or the project is already configured (scenario: user looks at a project the first time and doesn't know if secrets are already installed).

`isActionSupported` function only checks that the selected item is a suitable .NET project file, it is not about the whole validation flow. Action will be visible, but not active until the next check will confirm that the project file is suitable to introduce new secrets id.

But if you prefer not to go with such verbose flow, I will change it to how it worked before.

Witik replied:
I don't agree with the reasoning about being able to check if its working, as the same code is used for the other action.
The scenario you described is also not relevant as the other action will be visible when the project supports user secrets, and will be disabled when its not yet configured.

On the other hand, I DO agree with showing the disabled action to inform the user of its existence. So I accept the current proposed solution to show both actions when its a project file, and disable the open action when there are no secrets, and the initialize action when they are added.

Witik commented on src/main/resources/META-INF/plugin.xml:
```diff
@@ -4,17 +4,23 @@
 
     <description><![CDATA[
-      Open the user secrets for a project, for more information see the <a href="https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets">User Secrets documentation</a>.
+      Adds ability to Add new and to Open existing User Secrets, for more information see the <a href="https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets">User Secrets documentation</a>.
```
```suggestion
      Adds the ability to create and open User Secrets, for more information see the <a href="https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets">User Secrets documentation</a>.
```

seclerp replied:
Done

Witik commented on src/main/resources/META-INF/plugin.xml:
```diff
@@ -4,17 +4,23 @@
     <actions>
+        <action id="InitUserSecretsAction" class="eu.gillissen.rider.usersecrets.InitUserSecretsAction"
+                text="Initialize User Secrets"
+                description="Initializes user secrets for given project">
```
consistency
```suggestion
                description="Initializes user secrets for the project">
```

seclerp replied:
Done

Witik commented on src/main/kotlin/eu/gillissen/rider/usersecrets/UserSecretsService.kt:
```diff
@@ -0,0 +1,50 @@
+            .blockingGet(1, TimeUnit.MINUTES)
+            ?: return null
+
+        val secretsId = msBuildProperties[UserSecretsService.UserSecretsIdMsBuildProperty] ?: return null
+
+        return secretsId
```
Can be inlined, and elvis operator can be removed.
```suggestion
        return msBuildProperties[UserSecretsService.UserSecretsIdMsBuildProperty]
```

seclerp replied:
Done

Witik reviewed (changes requested):
Thanks for this PR! This will be a welcome addition to the plugin.

I left some comments with suggestions, please look them over.

seclerp commented:
@Witik thanks for the review! I fixed almost all suggestions, also added support of initializing user secrets for `Directory.Build.props` & `Directory.Build.targets` files, similar to what we have for the "Open user secrets" action. Please take a look at the changes and also at my concern about verbosity above.

Witik commented:
Thanks again! I'm going to test it on a few machines and then submit 1.2.0 for appoval!
